Spanner/src/SpannerClient.php

```diff
@@ -566,6 +578,14 @@ function (array $instance) {
      * $database = $spanner->connect('instance-id', 'database-id');
      * ```
      *
+     * Database role configured on the $options parameter
+     * will be applied to the session created by this object.
+     * Note: When the databseRole and sessionPool both are present in the options,
+     * we prioritize the sessionPool.
+     * ```
+     * $database = $spanner->connect('instance-id', 'database-id', ['databaseRole' => 'Reader']);
```

Comment:
Could you double check, it seems to expect a formatted version of `DatabaseRole`, like this [code pointer](https://github.com/googleapis/googleapis/blob/de1ace47a265f06f6d7da31f10f1679d685812a3/google/spanner/admin/database/v1/spanner_database_admin.proto#L888)

Reply:
And pls add a snippet test like this:

https://github.com/googleapis/google-cloud-php/blob/204f39fd8c82653da601dca367d2bee46451b0af/Spanner/tests/Snippet/SpannerClientTest.php#L192-L199

Reply:
The responsibility is to send the role in the session and the session variable defined like here: https://github.com/googleapis/google-cloud-php/blob/main/Spanner/src/V1/Session.php#L54

Thats where we came up with this way of sending the database role.

Just to clarify, you can find the database role to creator role conversion happening in each createSession calls in this PR.

Reply:
I will add the SnippetTest.

Reply:
Done